niworkflows/interfaces/bids.py

```diff
@@ -596,8 +596,16 @@ def _run_interface(self, runtime):
                         # load updated NIfTI
                         nii = nb.load(out_file, mmap=False)
                     data_dtype = np.dtype(data_dtype)
-                    if nii.get_data_dtype() != data_dtype:
+                    orig_dtype = nii.get_data_dtype()
+                    if orig_dtype != data_dtype:
+                        LOGGER.warning(
+                            f"Changing {out_file} dtype from {orig_dtype} to {data_dtype}"
+                        )
+                        # coerce dataobj to new data dtype
+                        new_data = np.asanyarray(nii.dataobj, dtype=data_dtype)
```

Comment:
Can't just cast to int without rounding.

```Python
>>> np.asanyarray(50.99999, np.int16)
array(50, dtype=int16)
```

What about:
```suggestion
                        if issubclass(data_dtype, np.integer):
                            new_data = np.rint(nii.dataobj, dtype=data_dtype)
                        else:
                            new_data = data_dtype(nii.dataobj)
```

Reply:
nice catch!